tests/_test_msui/test_mscolab.py

```diff
@@ -112,15 +119,17 @@ def test_connect_success(self, mockbox, mockset):
         assert mslib.utils.auth.get_password_from_keyring("MSCOLAB_AUTH_" + self.url, "mscolab") == "fnord"
         assert mockset.call_args_list == [mock.call("color: green;")]
 
-    def test_disconnect(self):
+    @mock.patch("PyQt5.QtWidgets.QMessageBox.question", return_value=QtWidgets.QMessageBox.No)
+    def test_disconnect(self, mockquestion):
         self._connect_to_mscolab()
         assert self.window.mscolab_server_url is not None
         QtTest.QTest.mouseClick(self.window.connectBtn, QtCore.Qt.LeftButton)
         assert self.window.mscolab_server_url is None
         # set ui_name_winodw default
         assert self.main_window.usernameLabel.text() == 'User'
 
-    def test_login(self):
+    @mock.patch("PyQt5.QtWidgets.QMessageBox.question", return_value=QtWidgets.QMessageBox.No)
+    def test_login(self, mockquestion):
         self._connect_to_mscolab()
         self._login(self.userdata[0], self.userdata[2])
```

Comment:
I disagree with the mocking here, we shouldn't just "click no" and be done with it. The issue here, why the message box pops up at all, is that in the msui config used in the test the "MSS_auth" does not have `self.userdata[0]` set as the username/email for the mscolab server we connect to here. With my approach to this (https://github.com/Open-MSS/MSS/pull/1967/files#diff-e21b0ebd4cef7ab5bc164aa57ddeeea4eda08ce271aec38819fb3b539a21b9cbR123-R126) the message box disappears entirely. We could also test that the message box pops up as expected when there is this mismatch in the config, but then we would have to also add a `mockquestion.assert_called_once_with(<args>)` in every test. I think that is something we should just do once in a separate test case.

Reply:
Case in point: there was a message box appearing in `test_get_airspaces` saying there was no airspace data in the file. That made no sense. It led me to discover this bug: https://github.com/Open-MSS/MSS/pull/1967/files#diff-ce023988bd24cea5863ed80b6ef4058629448da2f64480cd8dc31f06c5186a57R254-R256

On the other hand, in `test_get_airspaces_missing_data` we expected exactly this message box, so there we should mock the message box but then also assert that we get what we expect: https://github.com/Open-MSS/MSS/pull/1967/files#diff-d76a934985b02e5c8cafb478f5e5221eacd80d770c69798e9370ebb2232d8f9dR206-R215

Reply:
Obviously this argument applies to most of the other locations where this message box is mocked here as well.

Reply:
that's why I thought when you incorporate changes of this into your branch you can decide this on the merge, afterwardds we close this PR and you can verify the other ideas.

Reply:
ok